### PR TITLE
Add pitch-bend to midi example

### PR DESCRIPTION
**Pull request by gaborhermann**
I was playing around with my midi controller and could get the pitch-bend controller working. Hope somebody would find this useful.

(Btw. thanks for creating & maintaining Overtone! I love it!)

### CODE REVIEW

File: src/overtone/examples/midi/keyboard.clj [hunk 1]
```diff
@@ -85,9 +86,18 @@
...
                       :finished {}})
         sustain* (atom 0)
+        pitch-bend* (atom 0)
         on-id (keyword (gensym "on-handler"))
         off-id (keyword (gensym "off-handler"))
-        cc-id (keyword (gensym "cc-handler"))]
+        cc-id (keyword (gensym "cc-handler"))
+        pitch-bend-id (keyword (gensym "pitch-bend-handler"))
+        bytes-to-short (fn [b1 b2]
+                         (let [bb (. ByteBuffer allocate 2)
```
Comment (gaborhermann) [inline src/overtone/examples/midi/keyboard.clj]: This is a workaround as I'm not good in bit operations. I'd be more glad with a cleaner solution (no Java).

**Comment by hlolli** [comment]
Looks good, the inst-player is starting to look like a very complete function. Give me few days to test your changes, I'm working on some Overtone stuff that I need to do before. Looking at the code it does look to me that you don't need to convert it to bytes at any point. If I understand the behaviour that you're looking for, then for midi number 0 (pitch bender all the way down) you'd pitch bend octave down, and for midi number 127 (pitch bender all the way up) you'd pitch bend up an octave, please correct me if I'm wrong.

Note that no midi device can send `data2` of higher value than 127, therefore `(mod (+ 64 data2) 128)` should be redundant, or at maximum `(min (+ 64 data2) 128)` to not go over that limit, but I'd leave that to the hardware. Also as far I understand all starndard pitch wheel midi hardware defaults to 64 when it's in the center position, meaning when you add 64 to the midi message you're limting the range for only pitch bending down from 127 -> 64 (I guess that's where the modulo comes into play here, so if you don't raise 64 to data2 you won't need mod).

If you want to take this step further and make the pitch bend less linear and more exponential, you could translate the midi messages of the pitch bender so that 127 will add 12 to the fundemental midi note-on, 64 adds 0 and 0 subtracts 12. I haven't tested this byte-array solution you provided here so I'm not sure if it's working linearly or exponantially, I'm not so good with binary manipulation.

**Comment by gaborhermann** [comment]
Thanks for looking at the code.

Based on this
https://www.midikits.net/midi_analyser/pitch_bend.htm
the pitch-bend should be a 14 bit number between 0 and 16383, where 8192 means no change, 0 means one whole tone down (i.e. 2 half tones down), 16383 means one whole tone up. So I don't want to go up/down an octave, but only a tone. It's better to have a float number (e.g. between -1.0 and 1.0) so that we can easily convert it to note with a certain pitch. We could easily multiply that with 2 to get whole tone pitch-bends or 12 to get octave pitch-bends.

That 14 bit number is encoded in data1 and data2, and we need to translate that to that float. In the above code I used Java ByteBuffer to get a short number out of two bytes.

Probably you're right about `(mod (+ 64 data2) 128)` being redundant. I'll see if I can get rid of it, and test it. Maybe I'll play around with bit operations to see if I can get the same results. (Btw. I'm using an AKAI MPK mini MKII to play around with this.)

I only did linear pitch-bend, but exponential sounds interesting. Again, once we translated data1 and data2 to floats between -1.0 and 1.0, it's just a matter of applying a function to it, to make it exponential.

**Comment by hlolli** [comment]
Actually looking shortly over your code again, then there's no need to worry about the exponential part as you're feeding the value to `midicps` and that's like free liner->exp operator
```
(midicps (+ note pitch-bend))
```
I have akai midi controller but it doesn't have pitch bending wheel, I think my flatmate has one. I'll let you know when I've tested it.

But isn't `data2` just an integer that you receive, convert it into a byte and back to integer? Would it possibly work without converting to bytes at all? I'm worried as this is an "example" that newcomers would trip out of the complexity. But I think when we clean this up, we could move this function to overtone.device.midi or overtone.misc and rethink the documentation of Overtone.

**Comment by gaborhermann** [comment]
It's not straightforward because `data1` and `data2` both have infos, and we need to combine them into one number.

I modified the example to use bitwise operations. Probably the previous version wasn't 100% correct (it could have been of by a little) but the sound worked ok. This should be correct now, based on the MIDI standard (and testing with my keyboard).

I agree it's not an easy example to newcomers, so maybe we could include it in the overtone.device.midi? At least the function `midi-pitch-bend-to-float`. What do you think?

(Just as a reference this was also useful, explaining the conversion: http://sites.uci.edu/camp2014/2014/04/30/managing-midi-pitchbend-messages/)

**Comment by hlolli** [comment]
Nice, it works smoothly on the pitch bend of my flatmates midi controller, the code also looks very readable and clear.

**Comment by gaborhermann** [comment]
Super, thanks for testing & reviewing!

**Pull request merged**